Source/Dafny/Compilers/Compiler-js.cs

```diff
@@ -2386,12 +2386,32 @@ protected class ClassWriter : IClassWriter {
         CreateNoWindow = true,
         UseShellExecute = false,
         RedirectStandardInput = true,
-        RedirectStandardOutput = false,
-        RedirectStandardError = false,
+        RedirectStandardOutput = true,
+        RedirectStandardError = true,
       };
 
       try {
         using var nodeProcess = Process.Start(psi);
+        nodeProcess.BeginErrorReadLine();
+        nodeProcess.BeginOutputReadLine();
+
+        void ProcessErrorData(object sender, DataReceivedEventArgs e) {
+          if (!nodeProcess.HasExited) {
+            Console.Error.WriteLine(e.Data);
+          } else {
+            Console.Error.Write(e.Data);
+          }
+        }
+
+        void ProcessOutputData(object sender, DataReceivedEventArgs e) {
+          if (!nodeProcess.HasExited) {
+            Console.Out.WriteLine(e.Data);
+          } else {
+            Console.Out.Write(e.Data);
+          }
+        }
```

Comment:
Remove the duplication by extracting a method

Reply:
Just to add to @cpitclaudel's suggestion: I wouldn't declare it as an inner method since you don't require any closures.

Reply:
I _did_ close on on `nodeProcess`. However, after reading the documentation, it looks like the closure's first argument is the process itself, it just needed to be casted. I removed the duplication by creating a private method,